js_modules/dagster-ui/packages/ui-core/src/selection/SelectionAutoCompleteInput.tsx

```diff
@@ -37,6 +41,31 @@ export const SelectionAutoCompleteInput = <T extends Record<string, string[]>, N
   linter,
   attributesMap,
 }: SelectionAutoCompleteInputProps<T, N>) => {
+  const trackEvent = useTrackEvent();
+
+  const trackSelection = useMemo(() => {
+    return debounce((selection: string) => {
+      const selectionLowerCase = selection.toLowerCase();
+      const hasBooleanLogic =
+        selectionLowerCase.includes(' or ') ||
+        selectionLowerCase.includes(' and ') ||
+        selectionLowerCase.includes(' not ') ||
+        selectionLowerCase.startsWith('not ');
```

Comment:
Guess you could do something like this to avoid looking through the whole string multiple times, in case it's a long string:

```
const tokens = selectionLowerCase.split(' ');
const tokenSet = new Set(tokens);
const hasBooleanLogic = tokenSet.has(...) || ... || tokens[0] === 'not';
```

Reply:
bit overkill, don't expect more than an order of magnitude for the string, but definitely can do that if we run into any issues. The more robust way FWIW (which I considered) is to actually use a visitor to visit the string. Otherwise it's possible that there's a string in quotes that has " or " in it. eg: `tag:"example or example"`. In this case the string doesn't have boolean logic but we'd think it does. Not really worried about this edge case though.